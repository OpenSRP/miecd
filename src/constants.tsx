import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { TFunction } from 'i18next';

// openSRP api endpoints
export const OPENSRP_SECURITY_AUTHENTICATE = '/security/authenticate';

// internal urls
export const CHILD_PATIENT_DETAIL = 'child_patient_detail';
export const PATIENT_DETAIL = 'patient_detail';
export const NBC_AND_PNC_URL = '/nbc_and_pnc';
export const HIERARCHICAL_DATA_URL = '/hierarchicaldata';
export const PREGNANCY_URL = '/pregnancy';
export const LOGIN_URL = '/login';
export const LOGOUT_URL = '/logout';
export const HOME_URL = '/';
export const PREGNANCY_COMPARTMENTS_URL = '/pregnancy_compartments';
export const NBC_AND_PNC_COMPARTMENTS_URL = '/nbc_and_pnc_compartments';
export const CLIENT_URL = '/clients';
export const LOGFACE_URL = '/log-face';
export const PREGNANCY_LOGFACE_URL = '/pregnancy_log_face';
export const NBC_AND_PNC_LOGFACE_URL = '/nbc_and_pnc_logface_url';
export const NUTRITION_LOGFACE_URL = '/nutrition_logface_url';
export const NUTRITION_COMPARTMENTS_URL = '/nutrition_commpartments_url';
export const PREGNANCY_ANALYSIS_URL = '/pregnancy_analysis';
export const NBC_AND_PNC_ANALYSIS_URL = '/nbc_and_pnc_analysis';
export const NUTRITION_ANALYSIS_URL = '/nutrition_analysis';
export const NUTRITION_URL = '/nutrition';
export const HOUSEHOLD_URL = '/404';
export const ANC_URL = '/404';
export const CHILD_URL = '/404';
export const USER_URL = '/404';
export const ROLE_URL = '/404';
export const TEAM_URL = '/404';
export const LOCATIONS_URL = '/404';
export const REPORTS_URL = '/';
export const PATIENT_DETAIL_URL = `/${PATIENT_DETAIL}/:patient_id`;
export const CHILD_PATIENT_DETAIL_URL = `/${CHILD_PATIENT_DETAIL}/:patient_id`;
export const VIETNAM_COUNTRY_LOCATION_ID = 'd1865325-11e6-4e39-817b-e676c1affecf';
export const BACKEND_CALLBACK_URL = '/fe/oauth/callback/opensrp';
export const BACKEND_CALLBACK_PATH = '/fe/oauth/callback/:id';
export const REACT_CALLBACK_PATH = '/oauth/callback/:id';
export const BACKEND_LOGIN_URL = '/fe/login';
export const EXPRESS_LOGIN_URL = '/login';
export const REACT_LOGIN_URL = '/login';
export const EXPRESS_TOKEN_REFRESH_URL = '/refresh/token';

// string literals
export const HIGH = 'high';
export const LOW = 'low';
export const RED = 'red';
export const NO = 'no';
export const OVERWEIGHT = 'overweight';
export const STUNTED = 'stunted';
export const SEVERE_WASTING = 'severe wasting';
export const INAPPROPRIATELY_FED = 'inappropriately fed';
export const NOT_SET_LOWERCASE = 'not set';
export const PREGNANCY = 'Pregnancy';
export const NBC_AND_PNC = 'NBC & PNC';
export const NBC_AND_PNC_CHILD = 'NBC & PNC_CHILD';
export const NBC_AND_PNC_WOMAN = 'NBC & PNC_WOMAN';
export const NUTRITION = 'Nutrition';
export const NO_UNDERSCORE_RISK = 'no_risk';
export const RED_ALERT = 'red_alert';
export const NORMAL = 'normal';

// module string literals
export const PREGNANCY_MODULE = 'Pregnancy';
export const NUTRITION_MODULE = 'Nutrition';
export const NBC_AND_PNC_MODULE = 'NBC & PNC';

export const ALL = 'all';
export const UP = 'up';
export const NO_RISK_LOWERCASE = 'no risk';
export const NO_RISK = 'No Risk';
export const VIETNAM = 'Vietnam';
export const COUNTRY = 'Country';
export const PROVINCE = 'Province';
export const COMMUNE = 'Commune';
export const DISTRICT = 'District';
export const VILLAGE = 'Village';
export const RED_ALERT_CLASSNAME = 'red-alert';
export const NEWBORN_REPORT = 'Newborn Report';
export const EC_WOMAN = 'ec_woman';
export const EC_FAMILY_MEMBER = 'ec_family_member';
export const EC_CHILD = 'ec_child';

// smsTypes
export const PREGNANCY_DETECTION = 'Pregnancy Detection';
export const PREGNANCY_IDENTIFICATION = 'Pregnancy Identification';
export const PREGNANCY_REGISTRATION = 'Pregnancy Registration';
export const ANC_REPORT = 'ANC Report';
export const HOME_VISIT_REPORT = 'Home Visit Report';
export const SOCIAL_DETERMINANTS = 'Social Determinants';
export const DELIVERY_PLANNING = 'Delivery Planning';
export const BIRTH_REPORT = 'Birth Report';
export const DEATH_REPORT = 'Death Report';
export const RED_ALERT_REPORT = 'Red Alert Report';
export const RESPONSE_REPORT = 'Response Report';
export const REFUSAL_REPORT = 'Refusal Report';
export const DEPARTURE_CODE = 'Departure Code';
export const POSTNATAL_AND_NEWBORN_CARE = 'Postnatal and Newborn Care';
export const NUTRITION_REGISTRATION = 'Nutrition Registration';
export const MONTHLY_NUTRITION_REPORT = 'Monthly Nutrition Report';
export const ACCOUNT_CHECK = 'Account Check';
export const ANC_VISIT = 'ANC Visit';

// // sms events fields
export const NUTRITION_STATUS = 'nutrition_status';
export const GROWTH_STATUS = 'growth_status';
export const FEEDING_CATEGORY = 'feeding_category';
export const EVENT_ID = 'event_id';
export const LOGFACE_RISK = 'logface_risk';
export const HEIGHT = 'height';
export const WEIGHT = 'weight';
export const RISK = 'risk';
export const NUTRITION_REPORT = 'Nutrition Report';
export const RISK_LEVEL = 'risk_level';

// TIME constants
export const MICROSECONDS_IN_A_WEEK = 604800000;
export const GESTATION_PERIOD = 24192000000;

// Back arrow constant
export const BACKPAGE_ICON: IconProp = ['fas', 'arrow-left'];

// monthNames
export const getMonthNames = (t: TFunction) => [
    t('January'),
    t('February'),
    t('March'),
    t('April'),
    t('May'),
    t('June'),
    t('July'),
    t('August'),
    t('September'),
    t('October'),
    t('November'),
    t('December'),
];

// operational constants
export const DEFAULT_NUMBER_OF_LOGFACE_ROWS = 5;
export const EN_LANGUAGE_CODE = 'en' as const;
export const VI_LANGUAGE_CODE = 'vi' as const;

export const ALL_LANGUAGE_OPTIONS = {
    [EN_LANGUAGE_CODE]: 'English',
    [VI_LANGUAGE_CODE]: 'Vietnamese',
};

export const DATE_FORMAT = 'dd/MM/yyyy';
export const EVENT_DATE_DATE_FORMAT = 'yyyy-MM-dd';
export const SMS_MESSAGE_DATE_DATE_FORMAT = 'dd-MM-yyyy';
export const LOCATION_FILTER_PARAM = 'locationSearch';
export const RISK_CATEGORY_FILTER_PARAM = 'riskCategory';
export const SMS_TYPE_FILTER_PARAM = 'smsType';
export const SEARCH_FILTER_PARAM = 'search';

// react-query queryKey's
// compartment slices
export const FETCH_COMPARTMENTS_PREGNANCY_SLICE = 'FETCH_COMPARTMENTS_PREGNANCY_SLICE';
export const FETCH_COMPARTMENTS_NBC_AND_PNC_SLICE = 'FETCH_COMPARTMENTS_NBC_AND_PNC_SLICE';
export const FETCH_COMPARTMENTS_NUTRITION_SLICE = 'FETCH_COMPARTMENTS_NUTRITION_SLICE';
// location slices
export const FETCH_VILLAGES = 'FETCH_VILLAGES';
export const FETCH_COMMUNES = 'FETCH_COMMUNES';
export const FETCH_DISTRICTS = 'FETCH_DISTRICTS';
export const FETCH_PROVINCES = 'FETCH_PROVINCES';
// user location slice
export const FETCH_USER_LOCATION = 'FETCH_USER_LOCATION';
// user id
export const FETCH_USER_ID = 'FETCH_USER_ID';
