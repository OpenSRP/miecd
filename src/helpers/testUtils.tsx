import { mount, MountRendererProps } from 'enzyme';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import i18n from '../mls';
import { act } from 'react-dom/test-utils';

export const mountWithTranslations: typeof mount = (component: React.ReactNode, options?: MountRendererProps) => {
    return mount(<I18nextProvider i18n={i18n}>{component}</I18nextProvider>, options);
};

/**
 * Utility to await promises to resolve in tests
 * @param ms time in ms
 */
export async function waitForPromises() {
    await act(async () => {
        await new Promise((res) => setImmediate(res));
    });
}
