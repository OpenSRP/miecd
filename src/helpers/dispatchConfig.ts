import { ConfigState, setAllConfigs, getAllConfigs } from '@opensrp/pkg-config';
import { OPENSRP_REST_API_BASE_URL } from 'configs/settings';
import { BACKEND_ACTIVE } from '../configs/env';
import { BACKEND_LOGIN_URL, REACT_LOGIN_URL } from '../constants';
import i18n from '../mls';

export const APP_LOGIN_URL = BACKEND_ACTIVE ? BACKEND_LOGIN_URL : REACT_LOGIN_URL;

const defaultvalues = getAllConfigs();

const configObject: ConfigState = {
    ...defaultvalues,
    appLoginURL: APP_LOGIN_URL,
    opensrpBaseURL: OPENSRP_REST_API_BASE_URL,
    i18n,
};

setAllConfigs(configObject);
