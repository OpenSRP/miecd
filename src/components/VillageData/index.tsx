import React from 'react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import { Card, CardBody, CardTitle, Row, Table } from 'reactstrap';
import { NBC_AND_PNC_CHILD, NUTRITION, PREGNANCY } from '../../constants';
import { getModuleLink, getNumberOfDaysSinceDate } from '../../helpers/utils';
import { SmsData } from '../../store/ducks/sms_events';
import { PaginationData, Paginator, PaginatorProps } from '../Paginator';
import RiskColoring from '../RiskColoring';
import './index.css';

export interface Props {
    current_level: number;
    smsData: SmsData[];
    module: string;
}

interface State {
    currentPage: number;
}

const defaultProps: Props = {
    current_level: 0,
    module: '',
    smsData: [],
};

export type VillageDataPropsType = Props & WithTranslation;

class VillageData extends React.Component<VillageDataPropsType, State> {
    public static defaultProps = defaultProps;

    constructor(props: VillageDataPropsType) {
        super(props);
        this.state = {
            currentPage: 1,
        };
    }

    public render() {
        const { t } = this.props;

        const routePaginatorProps: PaginatorProps = {
            endLabel: t('last'),
            nextLabel: t('next'),
            onPageChange: (paginationData: PaginationData) => {
                this.setState({
                    currentPage: paginationData.currentPage,
                });
            },
            pageLimit: 5,
            pageNeighbours: 3,
            previousLabel: t('previous'),
            startLabel: t('first'),
            totalRecords: this.props.smsData.length,
        };

        return (
            <>
                {this.props.current_level >= 3 ? (
                    <>
                        <Row className="village villageDataRow">
                            <Card className="table-card">
                                <CardTitle>{t('Selected Commune')}</CardTitle>
                                <CardBody>
                                    <Table striped borderless>
                                        <thead id="header">
                                            {this.props.module === PREGNANCY ? (
                                                <tr>
                                                    <th className="default-width">{t('Patient ID')}</th>
                                                    <th className="default-width">{t('Gravidity')}</th>
                                                    <th className="default-width">{t('Parity')}</th>
                                                    <th className="default-width">{t('Location')}</th>
                                                    <th className="default-width">{t('EDD')}</th>
                                                    <th className="default-width">{t('Previous Pregnancy Risk')}</th>
                                                    <th className="default-width">{t('Risk category')}</th>
                                                </tr>
                                            ) : this.props.module === NUTRITION ? (
                                                <tr>
                                                    <th className="default-width">{t('Patient ID')}</th>
                                                    <th className="default-width">{t('Days since birth')}</th>
                                                    <th className="default-width">{t('Location of Residence')}</th>
                                                    <th className="default-width">{t('Current weight')}</th>
                                                    <th className="default-width">{t('Location of birth')}</th>
                                                    <th className="default-width">{t('Risk category')}</th>
                                                </tr>
                                            ) : (
                                                <tr>
                                                    <th className="default-width">{t('Patient ID')}</th>
                                                    <th className="default-width">{t('Days since birth')}</th>
                                                    <th className="default-width">{t('Location of Residence')}</th>
                                                    <th className="default-width">{t('Current symptoms')}</th>
                                                    <th className="default-width">{t('Location of birth')}</th>
                                                    <th className="default-width">{t('Risk category')}</th>
                                                </tr>
                                            )}
                                        </thead>
                                        <tbody id="body">
                                            {this.props.smsData.length
                                                ? this.props.smsData
                                                      .slice(
                                                          (this.state.currentPage - 1) * routePaginatorProps.pageLimit,
                                                          (this.state.currentPage - 1) * routePaginatorProps.pageLimit +
                                                              routePaginatorProps.pageLimit,
                                                      )
                                                      .map(
                                                          this.props.module === PREGNANCY
                                                              ? this.pregnancyMapFunction
                                                              : this.props.module === NBC_AND_PNC_CHILD
                                                              ? this.nbcAndPncChildMapFunction
                                                              : this.props.module === NUTRITION
                                                              ? this.nutritionMapFunction
                                                              : this.nbcAndPncMotherMapFunction,
                                                      )
                                                : null}
                                        </tbody>
                                    </Table>
                                </CardBody>
                            </Card>
                        </Row>
                        <Row id="navrow" className="villageDataRow">
                            <Paginator {...routePaginatorProps} />
                        </Row>
                    </>
                ) : null}
            </>
        );
    }

    /**
     * Returns a <tr></tr> for an SmsData object passed to it
     * which is used to build the table above.
     * for the NBC & PNC_MOTHER module
     * @param {SmsData} dataItem - an SmsData object used to generate a table row
     * @return {JSX.Element} table row
     */
    public nbcAndPncMotherMapFunction = (dataItem: SmsData): JSX.Element => {
        return (
            <tr key={dataItem.event_id}>
                <td className="default-width">
                    <Link to={`${getModuleLink(this.props.module)}/patient_detail/${dataItem.anc_id}`}>
                        {dataItem.anc_id}
                    </Link>
                </td>
                <td className="default-width">{getNumberOfDaysSinceDate(dataItem.EventDate)}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">{dataItem.mother_symptoms}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">
                    <RiskColoring {...{ risk: dataItem.logface_risk }} />
                </td>
            </tr>
        );
    };

    /**
     * Returns a <tr></tr> for an SmsData object passed to it
     * which is used to build the table above.
     * for the NBC & PNC_CHILD module
     * @param {SmsData} dataItem - an SmsData object used to generate a table row
     * @return {JSX.Element} table row
     */
    public nbcAndPncChildMapFunction = (dataItem: SmsData): JSX.Element => {
        return (
            <tr key={dataItem.event_id}>
                <td className="default-width">
                    <Link to={`${getModuleLink(this.props.module)}/child_patient_detail/${dataItem.anc_id}`}>
                        {dataItem.anc_id}
                    </Link>
                </td>
                <td className="default-width">{getNumberOfDaysSinceDate(dataItem.EventDate)}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">{dataItem.child_symptoms}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">
                    <RiskColoring {...{ risk: dataItem.logface_risk }} />
                </td>
            </tr>
        );
    };

    /**
     * Returns a <tr></tr> for an SmsData object passed to it
     * which is used to build the table above.
     * for the PREGNANCY module
     * @param {SmsData} dataItem - an SmsData object used to generate a table row
     * @return {JSX.Element} table row
     */
    public pregnancyMapFunction = (dataItem: SmsData): JSX.Element => {
        return (
            <tr key={dataItem.event_id}>
                <td className="default-width">
                    <Link to={`${getModuleLink(this.props.module)}/patient_detail/${dataItem.anc_id}`}>
                        {dataItem.anc_id}
                    </Link>
                </td>
                <td className="default-width">{dataItem.gravidity}</td>
                <td className="default-width">{dataItem.parity}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">{dataItem.lmp_edd}</td>
                <td className="default-width">{dataItem.previous_risks}</td>
                <td className="default-width">
                    <RiskColoring {...{ risk: dataItem.logface_risk }} />
                </td>
            </tr>
        );
    };

    /**
     * Returns a <tr></tr> for an SmsData object passed to it
     * which is used to build the table above.
     * for the NUTRITION module
     * @param {SmsData} dataItem - an SmsData object used to generate a table row
     * @return {JSX.Element} table row
     */
    public nutritionMapFunction = (dataItem: SmsData): JSX.Element => {
        return (
            <tr key={dataItem.event_id}>
                <td className="default-width">
                    <Link to={`${getModuleLink(this.props.module)}/child_patient_detail/${dataItem.anc_id}`}>
                        {dataItem.anc_id}
                    </Link>
                </td>
                <td className="default-width">{getNumberOfDaysSinceDate(dataItem.EventDate)}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">{dataItem.weight}</td>
                <td className="default-width">{dataItem.health_worker_location_name}</td>
                <td className="default-width">
                    <RiskColoring {...{ risk: dataItem.logface_risk }} />
                </td>
            </tr>
        );
    };
}

export default withTranslation()(VillageData);
