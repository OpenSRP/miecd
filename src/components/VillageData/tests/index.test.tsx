import { NBC_AND_PNC_WOMAN, NUTRITION, NBC_AND_PNC_CHILD, PREGNANCY } from '../../../constants';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import VillageData, { Props } from '..';
import { mountWithTranslations } from '../../../helpers/testUtils';
import store from '../../../store/index';
import {
    villageDataProps,
    pregnancyModuleData,
    nutritionModuleData,
    nbcPncChildModuleData,
    nbcPncMotherModuleData,
} from './villageDataPropsfixtures';
import MockDate from 'mockdate';

global.fetch = require('jest-fetch-mock');
MockDate.set('2021-04-12T19:31:00.000Z'); // 7-13-17 19:31 => Mersenne primes :)

const history = createBrowserHistory();

describe('components/VillageData', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });

    // eslint-disable-next-line jest/expect-expect
    it('must render without crashing', () => {
        shallow(<VillageData />);
    });

    it('must render correctly with data', () => {
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <VillageData {...(villageDataProps as Props)} />
                </Router>
            </Provider>,
        );
        wrapper.find('tbody#body tr').forEach((tr) => {
            expect(tr.text()).toMatchSnapshot('VillageData with data');
        });
        wrapper.unmount();
    });

    it('pagination works correctly', () => {
        const props = {
            ...villageDataProps,
            module: NUTRITION,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <VillageData {...(props as Props)} />
                </Router>
            </Provider>,
        );

        expect(wrapper.find('.pagination-container').text()).toMatchInlineSnapshot(`"previous12next"`);

        // inspect entries shown on page 1
        wrapper.find('tbody#body tr').forEach((tr) => {
            expect(tr.text()).toMatchSnapshot('first Page');
        });

        // go to page 2
        const page2PaginationItem = wrapper.find('.pagination-container ul li').at(3);
        expect(toJson(page2PaginationItem)).toMatchSnapshot('paginationItem for page 2');

        page2PaginationItem.simulate('click');
        wrapper.update();

        // inspect entries shown on page 2
        wrapper.find('tbody#body tr').forEach((tr) => {
            expect(tr.text()).toMatchSnapshot('second Page');
        });
        wrapper.unmount();
    });

    it('works correctly for NUTRITION', () => {
        const props = {
            ...villageDataProps,
            module: NUTRITION,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <VillageData {...(props as Props)} />
                </Router>
            </Provider>,
        );

        expect(wrapper.find('.pagination-container').text()).toMatchInlineSnapshot(`"previous12next"`);

        wrapper.find('tr').forEach((tr) => {
            expect(tr.text()).toMatchSnapshot('Nutrition Page');
        });
        wrapper.unmount();
    });

    it('works correctly for PREGNANCY', () => {
        const props = {
            ...villageDataProps,
            module: PREGNANCY,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <VillageData {...(props as Props)} />
                </Router>
            </Provider>,
        );

        expect(wrapper.find('.pagination-container').text()).toMatchInlineSnapshot(`"previous12next"`);

        wrapper.find('tr').forEach((tr) => {
            expect(tr.text()).toMatchSnapshot('Pregnancy Page');
        });
        wrapper.unmount();
    });

    it('works correctly for NBC_AND_PNC_CHILD', () => {
        const props = {
            ...villageDataProps,
            module: NBC_AND_PNC_CHILD,
            current_level: 2,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <VillageData {...(props as Props)} />
                </Router>
            </Provider>,
        );

        expect(wrapper.find('.pagination-container').text()).toMatchInlineSnapshot(`"previous12next"`);

        wrapper.find('tr').forEach((tr) => {
            expect(tr.text()).toMatchSnapshot('NBC_AND_PNC_CHILD page');
        });
        wrapper.unmount();
    });

    it('works correctly for NBC_AND_PNC_WOMAN', () => {
        const props = {
            ...villageDataProps,
            module: NBC_AND_PNC_WOMAN,
        };
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <VillageData {...(props as Props)} />
                </Router>
            </Provider>,
        );

        expect(wrapper.find('.pagination-container').text()).toMatchInlineSnapshot(`"previous12next"`);

        wrapper.find('tr').forEach((tr) => {
            expect(tr.text()).toMatchSnapshot('nBC and PNC woman Page');
        });
        wrapper.unmount();
    });
});

describe('components/VillageData/pregnancyMapFunction', () => {
    it('must return the correct value given specific input', () => {
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <VillageData {...(villageDataProps as Props)} />
                </Router>
            </Provider>,
        );
        const instance = wrapper.find('VillageData').instance() as typeof VillageData;
        const tableRows = shallow(instance.pregnancyMapFunction(pregnancyModuleData));
        const tableData = tableRows.find('td[className="default-width"]');
        expect(tableData.length).toEqual(8);
        // expect Patient ID row to match Patient ID
        const PatientIDRow = tableData.at(0).text();
        expect(PatientIDRow).toMatchInlineSnapshot(`"101CN2"`);
    });
});

describe('components/VillageData/nutritionMapFunction', () => {
    it('must return the correct value given specific input', () => {
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <VillageData {...(villageDataProps as Props)} />
                </Router>
            </Provider>,
        );
        const instance = wrapper.find('VillageData').instance() as typeof VillageData;
        const tableRows = shallow(instance.nutritionMapFunction(nutritionModuleData));
        const tableData = tableRows.find('td[className="default-width"]');
        expect(tableData.length).toEqual(4);
        // expect Patient ID row to match Patient ID
        const PatientIDRow = tableData.at(0).text();
        expect(PatientIDRow).toMatchInlineSnapshot(`"100TPL"`);
    });
});

describe('components/VillageData/nbcAndPncChildMapFunction', () => {
    it('must return the correct value given specific input', () => {
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <VillageData {...(villageDataProps as Props)} />
                </Router>
            </Provider>,
        );
        const instance = wrapper.find('VillageData').instance() as typeof VillageData;
        const tableRows = shallow(instance.nbcAndPncChildMapFunction(nbcPncChildModuleData));
        const tableData = tableRows.find('td[className="default-width"]');
        expect(tableData.length).toEqual(6);
        // expect Patient ID row to match Patient ID
        const PatientIDRow = tableData.at(0).text();
        expect(PatientIDRow).toMatchInlineSnapshot(`"100T5T-20190920-01"`);
    });
});

describe('components/VillageData/nbcAndPncMotherMapFunction', () => {
    it('must return the correct value given specific input', () => {
        const wrapper = mountWithTranslations(
            <Provider store={store}>
                <Router history={history}>
                    <VillageData {...(villageDataProps as Props)} />
                </Router>
            </Provider>,
        );
        const instance = wrapper.find('VillageData').instance() as typeof VillageData;
        const tableRows = shallow(instance.nbcAndPncMotherMapFunction(nbcPncMotherModuleData));
        const tableData = tableRows.find('td[className="default-width"]');
        expect(tableData.length).toEqual(7);
        // expect Patient ID row to match Patient ID
        const PatientIDRow = tableData.at(0).text();
        expect(PatientIDRow).toMatchInlineSnapshot(`"100VNL"`);
    });
});
