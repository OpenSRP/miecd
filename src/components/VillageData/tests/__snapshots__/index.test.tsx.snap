// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/VillageData must render correctly with data: VillageData with data 1`] = `"101CN224 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious Miscarriage/ Stillborn / Pregnancy termination2021-10-01Not at home/facilities, ENROUTERisk"`;

exports[`components/VillageData must render correctly with data: VillageData with data 2`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-section2021-06-17At homeRisk"`;

exports[`components/VillageData must render correctly with data: VillageData with data 3`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-section2021-06-17At homeRisk"`;

exports[`components/VillageData must render correctly with data: VillageData with data 4`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-section2021-06-17At homeRisk"`;

exports[`components/VillageData must render correctly with data: VillageData with data 5`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-section2021-06-17At homeRed alert"`;

exports[`components/VillageData pagination works correctly: first Page 1`] = `"101CN224 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData pagination works correctly: first Page 2`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData pagination works correctly: first Page 3`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData pagination works correctly: first Page 4`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData pagination works correctly: first Page 5`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData pagination works correctly: paginationItem for page 2 1`] = `
<li
  className="page-item"
>
  <a
    aria-disabled="false"
    aria-label="Next page"
    className="page-link"
    onClick={[Function]}
    onKeyPress={[Function]}
    rel="next"
    role="button"
    tabIndex="0"
  >
    next
  </a>
</li>
`;

exports[`components/VillageData pagination works correctly: second Page 1`] = `"101CN224 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData pagination works correctly: second Page 2`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData pagination works correctly: second Page 3`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData pagination works correctly: second Page 4`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData pagination works correctly: second Page 5`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_CHILD: NBC_AND_PNC_CHILD page 1`] = `"Patient IDDays Since BirthLocation Of ResidenceCurrent SymptomsLocation Of BirthRisk Category"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_CHILD: NBC_AND_PNC_CHILD page 2`] = `"101CN2NaNThôn Kon Hia 1Risk"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_CHILD: NBC_AND_PNC_CHILD page 3`] = `"101C8YNaNThôn Kon Hia 1Risk"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_CHILD: NBC_AND_PNC_CHILD page 4`] = `"101C8YNaNThôn Kon Hia 1Risk"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_CHILD: NBC_AND_PNC_CHILD page 5`] = `"101C8YNaNThôn Kon Hia 1Risk"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_CHILD: NBC_AND_PNC_CHILD page 6`] = `"101C8YNaNThôn Kon Hia 1Red alert"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_WOMAN: nBC and PNC woman Page 1`] = `"Patient IDAgeLocation Of ResidenceCurrent SymptomsPrevious Risks / Existing Medical ConditionsLocation Of BirthRisk Category"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_WOMAN: nBC and PNC woman Page 2`] = `"101CN224 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious Miscarriage/ Stillborn / Pregnancy terminationRisk"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_WOMAN: nBC and PNC woman Page 3`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-sectionRisk"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_WOMAN: nBC and PNC woman Page 4`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-sectionRisk"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_WOMAN: nBC and PNC woman Page 5`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-sectionRisk"`;

exports[`components/VillageData works correctly for NBC_AND_PNC_WOMAN: nBC and PNC woman Page 6`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-sectionRed alert"`;

exports[`components/VillageData works correctly for NUTRITION: Nutrition Page 1`] = `"Patient IDAgeLocation Of ResidenceRisk Category"`;

exports[`components/VillageData works correctly for NUTRITION: Nutrition Page 2`] = `"101CN224 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData works correctly for NUTRITION: Nutrition Page 3`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData works correctly for NUTRITION: Nutrition Page 4`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData works correctly for NUTRITION: Nutrition Page 5`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData works correctly for NUTRITION: Nutrition Page 6`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1"`;

exports[`components/VillageData works correctly for PREGNANCY: Pregnancy Page 1`] = `"Patient IDAgeLocation Of ResidenceCurrent SymptomsPrevious Risks / Existing Medical ConditionsEDDPlanned Delivery LocationRisk Category"`;

exports[`components/VillageData works correctly for PREGNANCY: Pregnancy Page 2`] = `"101CN224 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious Miscarriage/ Stillborn / Pregnancy termination2021-10-01Not at home/facilities, ENROUTERisk"`;

exports[`components/VillageData works correctly for PREGNANCY: Pregnancy Page 3`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-section2021-06-17At homeRisk"`;

exports[`components/VillageData works correctly for PREGNANCY: Pregnancy Page 4`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-section2021-06-17At homeRisk"`;

exports[`components/VillageData works correctly for PREGNANCY: Pregnancy Page 5`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-section2021-06-17At homeRisk"`;

exports[`components/VillageData works correctly for PREGNANCY: Pregnancy Page 6`] = `"101C8Y33 Years 04 Months 18 DaysThôn Kon Hia 1CoughingPrevious C-section2021-06-17At homeRed alert"`;
